options.go

```diff
@@ -396,17 +402,53 @@ func (o *Options) Validate() error {
 		msgs = append(msgs, fmt.Sprintf("cookie_samesite (%s) must be one of ['', 'lax', 'strict', 'none']", o.CookieSameSite))
 	}
 
+	msgs = parseClaimAuthorizations(o, msgs)
 	msgs = parseSignatureKey(o, msgs)
 	msgs = validateCookieName(o, msgs)
 	msgs = setupLogger(o, msgs)
 
+	// Create a key that's unique to the configured instance of the proxy. If this changes, all
+	// existing signed cookies (and thus their sessions) will be considered invalid and the user
+	// will be forced to re-authenticate (even if the provider does so silently). The CookieSecret
+	// is a key bit of salt in the process.
+	hmacKey := append(o.claimsAuthorizer.RulesHash(), o.CookieSecret...)
+	for _, domain := range o.EmailDomains {
+		hmacKey = append(hmacKey, domain...)
+	}
+	hmacKey, msgs = applyFileDigestSalt(hmacKey, msgs, o.AuthenticatedEmailsFile)
+	hmacKey, msgs = applyFileDigestSalt(hmacKey, msgs, o.HtpasswdFile)
+	hmacKey = o.provider.ApplyConfigSalt(hmacKey)
+	o.CookieHmacKey = hmacKey
```

Comment:
What happens if you are running multiple instances of the OAuth2 Proxy in an HA set up, this looks like it currently prevents the proxy from being horizontally scalable?

Reply:
Well, assuming the configs deployed to each are identical, I believe it should be fine. There are no non-deterministic elements in the list. (Though if a provider implementation added one, that'd be an issue).

If the configs differ, that's another issue. But then, it'd be weird to have HA versions having different configs, no?

Reply:
I can describe a not-too-uncommon example (which isn't exactly HA): Multiple services, on separate sub-domains, on separate servers, using separate oauth2-proxy instances, but sharing the cookie domain ".toplevel.com". The configs have different upstreams and different callback URLs, so signing in at one is specific to that one, but after doing that you're also signed into all the others.

If they have different claims requirements (groups, or email list, etc) then that may not work right ... what can be checked on every request vs only at sign_in currently depends on implementation details I think ...

Anyway, if the set of config settings hashed into this cookie signature does not include upstream and callback redirect, then it would be permissive enough for the simple multi-subdomain multi-oauth2-proxy case. (So I think this PR is currently OK for that.)